stanford-sw/src/edu/stanford/StanfordIndexer.java

```diff
@@ -2380,12 +2438,47 @@ private void addSDRfrom856s(final Record record) {
     Set<String> resultSet = new LinkedHashSet<String>();
     for (String val : MarcUtils.getFieldList(record, fieldSpec)) {
         String result = Utils.removeAllTrailingCharAndPeriod(val, "(" + charsToReplaceRegEx + ")+", charsB4periodRegEx);
+        result = cleanFacetPunct(result);
       resultSet.add(result);
     }
 
     return resultSet;
   }
 
+    public String cleanFacetPunct(String origStr)
```

Comment:
yes, it is what I meant, and yes, it is okay.  All this code is, essentially, ours.  It will probably never been merged upstream, and at best we would do some cut and paste to get upstream changes, should we ever want them.